src/modeler/AutoRest.Swagger/Validation/BodyTopLevelProperties.cs

```diff
@@ -70,11 +52,37 @@ public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<
         /// </summary>
         public override Category Severity => Category.Error;
 
-
         /// <summary>
         /// What kind of change implementing this rule can cause.
         /// </summary>
         public override ValidationChangesImpact ValidationChangesImpact => ValidationChangesImpact.ServiceImpactingChanges;
 
+        /// <summary>
+        /// This rule passes if the model definition contains top level properties only from the allowed set: name, type,
+        /// id, location, properties, tags, plan, sku, etag, managedBy, identity
+        /// </summary>
+        /// <param name="definitions">The model definitions</param>
+        /// <param name="context">The context object</param>
+        /// <returns>validation messages</returns>
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Schema> definitions, RuleContext context)
+        {
+            foreach(string resourceModel in context.ResourceModels)
+            {
+                IEnumerable<KeyValuePair<string, Schema>> topLevelProperties = ValidationUtilities.EnumerateProperties(resourceModel, definitions);
+                IEnumerable<KeyValuePair<string, Schema>> violatingProperties = topLevelProperties.Where(topLevelProperty => !AllowedTopLevelProperties.Contains(topLevelProperty.Key));
+
+                if (violatingProperties != null && violatingProperties.Count() != 0)
+                {
+                    List<string> list = new List<string>();
```

Comment:
3 lines turn to one thanks to linq
```var list = violatingProperties.Select(prop=>prop.Key);```

On a different note, this will indicate a problem with a property not directly tied to the current model,
Should we also figure out what model the property belongs to when reporting the validation message?